mlir/lib/IR/Builders.cpp

```diff
@@ -537,7 +538,7 @@ Operation *OpBuilder::clone(Operation &op, IRMapping &mapper) {
     for (Region &region : newOp->getRegions())
       region.walk(walkFn);
```

Comment:
Your change is changing the notification order for `newOp` from post-order to pre-order. For consistency, `region.walk` should also be a pre-order walk now.

Reply:
Thanks for catching this! Updated the walk to use pre-order and the test to verify that.